### DISCUSSION TITLE
Errors in the writing test stage for solidity smart contract

### DISCUSSION THREAD

**Discussion by mhmmadexsal** [Cyfrin/foundry-full-course-cu#1820] [General]
// SPDX-License-Identifier: SEE LICENSE IN LICENSE
pragma solidity ^0.8.19;

import {Test, console} from "forge-std/Test.sol";
import {FundMe} from "../src/FundMe.sol";

contract FundMeTest is Test {
    FundMe fundMe;

    function setUp() external {
        fundMe = new FundMe();
    }

    function testMinimumDollarIsFive() public {
        assertEq(fundMe.MINIMUM_USD(), 5e18);
    }
}

**Comment by mhmmadexsal** [comment]
forge test
Compiler run failed:
Error (6160): Wrong argument count for function call: 0 arguments given but expected 1.
  --> test/FundMeTest.t.sol:11:18:
   |
11 |         fundMe = new FundMe();
   |                  ^^^^^^^^^^^^

Error: 
Compilation failed

**Comment by urosognjenovic** [comment]
You have to pass the argument to the constructor:

```fundMe = new FundMe(priceFeed)```, where ```priceFeed``` is the address of the price feed.

**Reply by mhmmadexsal** [reply]
> You have to pass the argument to the constructor:
> 
> `fundMe = new FundMe(priceFeed)`, where `priceFeed` is the address of the price feed.

Compiler run failed:
Error (7576): Undeclared identifier.
  --> test/FundMeTest.t.sol:11:29:
   |
11 |         fundMe = new FundMe(priceFeed);
   |                             ^^^^^^^^^

Error: 
Compilation failed

**Reply by urosognjenovic** [reply]
Please share your repository with us so we can assist, otherwise we're just blindly guessing the cause of the error.

**Comment by EngrPips** [comment]
Hello @mhmmadexsal, Do as @thebraveman17  suggested and let us know if that resolves your issue.

**Comment by EngrPips** [comment]
@mhmmadexsal, are you following Patrick's course or trying to do this yourself? I asked because you should have a `helperConfig` and a `deployment script` that will properly deploy a `FundMe` contract if you have gotten to the test part of the section. I've shared my deployment script below.

<img width="1440" alt="Screenshot 2024-06-11 at 15 02 27" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/a45c3074-c118-4bb6-ae29-d862681db08a">

**Comment by Lukelr12** [comment]
I am having an issue with the first bit of test writing, I am getting this error 

Error (6160): Wrong argument count for function call: 0 arguments given but expected 1.
  --> test/fundMeTest.t.sol:12:18:

![Screenshot (1)](https://github.com/user-attachments/assets/0b405e50-c4fa-4247-95a1-c274d28ea7f0)

**Reply by EngrPips** [reply]
Hello @mhmmadexsal, Your `FundMe` contract takes one argument in the constructor, but you did not pass any argument when you did `fundMe = new FundMe()`

**Reply by Lukelr12** [reply]
Fixed

'''
 function setUp() external {
    fundMe = new FundMe(msg.sender); '''

Thank you again EngrPips!!

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it.
